### PR TITLE
Add pagination for scan and query

### PR DESCRIPTION
**Pull request by tgrk**
* integrated into Travis CI
* pagination for scan and query
* Minor doc updates about setup, dependencies etc

### CODE REVIEW

File: test/current_test.erl [hunk 1]
```diff
@@ -152,15 +154,32 @@ scan() ->
 
     ok = current:batch_write_item(Request, []),
 
-    Q = {[{<<"TableName">>, ?TABLE}]},
+    Q = {[{<<"TableName">>, ?TABLE},
+          {<<"Limit">>, 10}]},
```
Comment (chrisavl) [inline test/current_test.erl]: Why do we have {Limit, 10} here, shouldn't that actually break this test?

File: test/current_test.erl [hunk 2]
```diff
@@ -152,15 +154,32 @@ scan() ->
...
+                 current:scan(ErrorQ, [])),
 
+    %% Limit and pagging
+    {ok, LimitedItems1, LastEvaluatedKey1} = current:scan(Q, [{max_items, 80}]),
+    ?assertEqual(80, length(LimitedItems1)),
+
+    %% Pagging last page
+    Q1 = {[{<<"TableName">>, ?TABLE},
+           {<<"ExclusiveStartKey">>, LastEvaluatedKey1},
+           {<<"Limit">>, 10}]},
```
Comment (chrisavl) [inline test/current_test.erl]: {Limit, 10} again here.

File: test/current_test.erl [hunk 3]
```diff
@@ -261,10 +281,27 @@ q() ->
...
+    ?assertEqual(80, length(LimitedItems1)),
 
+    %% Pagging last page
+    Q1 = {[{<<"TableName">>, ?TABLE},
+          {<<"KeyConditions">>,
+           {[{<<"hash_key">>,
+              {[{<<"AttributeValueList">>, [{[{<<"N">>, <<"1">>}]}]},
+                {<<"ComparisonOperator">>, <<"EQ">>}]}}]}},
+           {<<"ExclusiveStartKey">>, LastEvaluatedKey1},
+           {<<"Limit">>, 10}]},
```
Comment (chrisavl) [inline test/current_test.erl]: Another Limit I don't understand :)

Comment (tgrk) [inline test/current_test.erl]: Copy'n paste error :)

File: src/current.erl [hunk 4]
```diff
@@ -346,46 +339,65 @@ do_query({UserRequest}, {Acc, AccLen}, Opts) ->
...
-
-
 %%
 %% INTERNALS
 %%
 
+update_query(Request, Key, Value) ->
+    lists:keystore(Key, 1, Request, {Key, Value}).
+
+prepare_query(UserRequest, Opts) ->
```
Comment (chrisavl) [inline src/current.erl]: We don't actually need this function, it just duplicates the ExclusiveStartKey key in the request if it is there already. Would be nice to get rid of this redundancy (it was also redundant in the original code, probably added by accident).

File: src/current.erl [hunk 5]
```diff
@@ -1,6 +1,9 @@
 %% @doc: DynamoDB client
 -module(current).
 
+%%FIXME: this is only for debugging!!!
+-include_lib("eunit/include/eunit.hrl").
```
Comment (chrisavl) [inline src/current.erl]: This can be removed right?

**Pull request merged**